src/transport/SecureSession.cpp

```diff
@@ -49,6 +50,27 @@ void SecureSession::MarkForRemoval()
     }
 }
 
+void SecureSession::MarkForInactive()
+{
+    ChipLogDetail(Inet, "SecureSession[%p]: MarkForInactive Type:%d LSID:%d", this, to_underlying(mSecureSessionType),
+                  mLocalSessionId);
+    ReferenceCountedHandle<Transport::Session> ref(*this);
+    switch (mState)
+    {
+    case State::kPairing:
+        VerifyOrDie(false);
+        return;
+    case State::kActive:
+        // By setting this state, IsActiveSession() will return false, which prevent creating new exchanges.
+        mState = State::kInactive;
+        return;
+    case State::kInactive:
+    case State::kPendingRemoval:
```

Comment:
This should be a `VerifyOrDie(false)` no? If someone is calling `MarkForInactive` on a session that is slated for eviction, that is a logic bug and should not happen.

Reply:
I'd agree that this should not happen now, but the constraint is hidden deeply in the application logic.

If a vendor is going to implement a custom command which do release a session and also remove the fabric. There is nothing in the spec to prevent he implementing this command, then the constraint becomes false.

Reply:
Lets do any more changes as followup. Resolving for now to merge as this PR has been open for a while and I am trying to close up large PRs as 1.0 branch date nears.